concordium-sdk/src/main/java/com/concordium/sdk/transactions/TransferToEncryptedTransaction.java

```diff
@@ -0,0 +1,95 @@
+package com.concordium.sdk.transactions;
+
+
+import com.concordium.sdk.exceptions.TransactionCreationException;
+import com.concordium.sdk.types.UInt64;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.val;
+
+import java.util.Objects;
+
+/**
+ * Construct a transaction to transfer from public to encrypted balance of the sender account.
+ */
+@Getter
+public class TransferToEncryptedTransaction extends AbstractTransaction {
+    /**
+     * The amount to transfer from public to encrypted balance of the sender account.
+     */
+    private final CCDAmount amount;
+
+    /**
+     * Account Address of the sender.
+     */
+    private final AccountAddress sender;
+    /**
+     * The senders account next available nonce.
+     */
+    private final AccountNonce nonce;
+    /**
+     * Indicates when the transaction should expire.
+     */
+    private final Expiry expiry;
+    /**
+     * A signer object that is used to sign the transaction.
+     */
+    private final TransactionSigner signer;
+
+    /**
+     * Maximum energy **allowed** for the transaction to use.
+     */
+    private final UInt64 maxEnergyCost;
```

Comment:
There shouldn't be a maxEnergyCost. 

Instead the cost should be calculated in the sdk just as with normal transfers. 

The actual calculation is a bit different due to `TRANSFER_TO_ENCRYPTED`. 
See https://github.com/Concordium/concordium-base/blob/75f015bb5f6483f5fdb5b49c789113a6f419038a/rust-src/concordium_base/src/transactions.rs#L1994